client/src/pages/private/ParticipantView.js

```diff
@@ -335,7 +335,8 @@ export default () => {
         setLoadingUser(false);
         setSites(sites);
         setRoles(roles);
-        
+        // a Boolean filter removes all undefined values
+        setLocations(roles.map((loc) => locationRoles[loc]).filter(Boolean));
```

Comment:
Curious why this was added? I thought the tests might need it but that doesn't seem to be the case :)

Reply:
More redundant code! I implemented a better version of this setter a few lines down.